### PR TITLE
Test travis

### PR DESCRIPTION
**Pull request by phadej**
Doesn't fail with osx, thought locally it does...

### CODE REVIEW

**Comment by ezyang** [comment]
Maybe you'll have better luck getting a more timely Travis build by asking Travis to do it on your home repo.

**Comment by phadej** [comment]
I can wait.

**Comment by phadej** [comment]
from log https://travis-ci.org/haskell/cabal/jobs/147287449

```sh
...
  RegisterAssumeDepsUpToDate:                   OK (2.17s)
    Initial build and copy
    Register q
    Register p
1 out of 77 tests failed (243.66s)
1 # <-- echo $?
```

Dunno why :/

**Comment by ezyang** [comment]
Your second test is reasonable but `travis-common.sh` is supposed to `set -e`, and my local test shows this is supposed to work.

**Comment by phadej** [comment]
it is. But shell scripting is total mystery to me.

Travis transforms `.travis.yml` parts into shell script, and I *don't* know if there is something top level which might interfere with our build shell scripts.

**Comment by phadej** [comment]
Finally. It fails (only) on osx. I'll remove the revert, and push soon.

File: travis-bootstrap.sh [hunk 1]
```diff
@@ -42,11 +42,11 @@ timed cabal update
 # issue, and it is easy enough to work around: clean first.
 
 echo Cabal
-(cd Cabal && timed cabal clean)
```
Comment (23Skidoo) [inline travis-bootstrap.sh]: Eww, shell sucks. Wish there was a nicer way to fix this.

Comment (phadej) [inline travis-bootstrap.sh]: Well, I have no idea why travis osx /bin/sh is /bin/bash, my osx has real `sh`, but I don't want to invest in this too much.

Comment (phadej) [inline travis-bootstrap.sh]: Alternatively we could use `bash` everywhere and `{ }` syntax, i.e. no subshells. But ah.

I was thinking of writing own shell scripting language just to make CI stuff easier, but the pain and frustration levels haven't risen enough yet.

Comment (23Skidoo) [inline travis-bootstrap.sh]: Well, you spent enough time on this already, so let's merge this patch in the current form. In the (distant) future maybe we can rewrite these scripts and `bootstrap.sh` in Haskell.

**Comment by ezyang** [comment]
@phadej, thank you for taking time to find a workaround. We need to somehow document why this workaround is needed in the code, less a bash-hacker looks and says, "Why, of course this `||` business is unnecessary, exit codes get inherited."

Indeed, the behavior we see here has nothing to do with Mac OS X using bash to interpret the shell script or not (which is actually quite common.) Look at the following litmus test:

```
#!/bin/bash
set -e
foo() {
    false
}
(cd . && foo)
echo not printed
```

This does not print anything on bash 4.3.46. I have heard that Mac OS X runs a very old version of bash, so it is possible that the syntax that I am using is tickling a bug.

**Comment by ezyang** [comment]
OK, I built a copy of bash-3.2 and verified that it indeed prints "not printed". So basically Mac OS X bash is too old :o)

**Comment by 23Skidoo** [comment]
@ezyang Thanks for investigating. I expanded the comment in `travis-script.sh` a bit.

**Comment by tuncer** [comment]
Just to be clear, is the plan to turn this into a bash script (again)?

**Comment by phadej** [comment]
@tuncer no. but on my OSX machine:
```
% /bin/sh --version            
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)
Copyright (C) 2007 Free Software Foundation, Inc.
```

**Comment by tuncer** [comment]
@phadej, the only problem that can cause is that one accidentally uses non-POSIX feature. However, as long as the CI env has a POSIX sh as `/bin/sh`, I wouldn't expect regressions.

**Pull request merged**